packages/allow-scripts/src/index.js

```diff
@@ -1,70 +1,231 @@
+const { linkBinAbsolute, linkBinRelative } = require('./linker.js')
+const { FT } = require('./toggles.js')
 const { loadCanonicalNameMap } = require('@lavamoat/aa')
 
+/**
+ * @typedef {Object} PkgConfs
+ * @property {Object} packageJson
+ * @property {Object} configs
+ * @property {ScriptsConfig} configs.lifecycle
+ * @property {BinsConfig} configs.bin
+ * @property {boolean} somePoliciesAreMissing
+ *
+ * Individual package info 
+ * @typedef {Object} PkgInfo
+ * @property {string} canonicalName
+ * @property {string} path
+ * @property {Object} scripts
+ *
+ * Individual bin link info 
+ * @typedef {Object} BinInfo
+ * @property {string} canonicalName
+ * @property {boolean} isDirect
+ * @property {string} bin
+ * @property {string} path
+ * @property {string} link
+ * @property {string} fullLinkPath
+ *
+ * Configuration for a type of scripts policies
+ * @typedef {Object} ScriptsConfig
+ * @property {Object} allowConfig
+ * @property {Map<string,[PkgInfo]>} packagesWithScripts
+ * @property {Array} allowedPatterns
+ * @property {Array} disallowedPatterns
+ * @property {Array} missingPolicies
+ * @property {Array} excessPolicies
+ *
+ * @typedef {Map<string,[BinInfo]>} BinCandidates 
+ *
+ * Configuration for a type of bins policies
+ * @typedef {Object} BinsConfig
+ * @property {Object} allowConfig
+ * @property {BinCandidates} binCandidates
+ * @property {Array<BinInfo>} allowedBins
+ * @property {Array<BinInfo>} firewalledBins
+ * @property {Array} excessPolicies
+ * @property {boolean} somePoliciesAreMissing
+ */
+
 module.exports = {
+  getOptionsForBin,
   runAllowedPackages,
   setDefaultConfiguration,
   printPackagesList,
 }
 
-async function runAllowedPackages ({ rootDir }) {
+async function getOptionsForBin({ rootDir, name }) {
   const {
-    packagesWithLifecycleScripts,
-    allowedPatterns,
-    missingPolicies
+    configs: {
+      bin: {
+        binCandidates
+      }
+    }
   } = await loadAllPackageConfigurations({ rootDir })
 
-  if (missingPolicies.length) {
+  return binCandidates.get(name)
+}
+
+
+async function runAllowedPackages({ rootDir }) {
+  const {
+    configs: {
+      lifecycle,
+      bin
+    },
+    somePoliciesAreMissing
+  } = await loadAllPackageConfigurations({ rootDir })
+
+  if (somePoliciesAreMissing) {
     console.log('\n@lavamoat/allow-scripts has detected dependencies without configuration. explicit configuration required.')
     console.log('run "allow-scripts auto" to automatically populate the configuration.\n')
 
-    console.log('packages missing configuration:')
-    missingPolicies.forEach(pattern => {
-      const collection = packagesWithLifecycleScripts.get(pattern) || []
-      console.log(`- ${pattern} [${collection.length} location(s)]`)
-    })
+    printMissingPoliciesIfAny(lifecycle)
 
     // exit with error
     process.exit(1)
   }
 
+  if (FT.bins && bin.allowConfig) {
+    // Consider: Might as well delete entire .bin and recreate in case it was left there
```

Comment:
I think this really shouldn't be taken lightly - modifying package manager rc files an package.json is one thing but doing this is something that should not be done without a capitalized "Are you sure?" or explicit `--force-delete-bins` flag... Users may have setups and expectations we have no idea about that would make this a bad surprise.

If anything: Recreate in a new temp location, diff with current, error/warn about diffs

Reply:
This note originates from the fact that yarn3 doesn't have a setting to disable bin linking and we'll have to do it for yarn3 somehow if we want full control over bin links at all times.
The other pkgmanagers get configured to leave the .bin empty and we're running with that assumption. Existing items in node_modules/.bin will get overwritten anyway (I explored an idea that'd put stuff there before running install, and all pkgmanagers destroyed what I created unless blocked on filesystem level requiring root access)

I'm considering this step as a backup in case there was a bypass in existing protections. if the assumption is .bin doesn't exist, deleting it preemtively seems like a good idea...

Reply:
Hm, if the case really is that all (at least supported) package managers wipe it regardless, I guess it should be all right.

Reply:
yes, in a future change, as we refine the logic against real usage.